src/components/application_manager/rpc_plugins/vehicle_info_plugin/test/vehicle_info_plugin_test.cc

```diff
@@ -0,0 +1,238 @@
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE. */
+
+#include "gtest/gtest.h"
+
+#include "application_manager/mock_application.h"
+#include "application_manager/mock_hmi_capabilities.h"
+#include "application_manager/mock_message_helper.h"
+#include "application_manager/mock_rpc_service.h"
+#include "application_manager/policies/mock_policy_handler_interface.h"
+#include "application_manager/smart_object_keys.h"
+#include "test/application_manager/mock_application_manager.h"
+
+namespace app_manager_test = test::components::application_manager_test;
+namespace vehicle_info_plugin_test {
+
+using ::testing::_;
+using ::testing::Mock;
+using ::testing::NiceMock;
+using ::testing::Return;
+using ::testing::ReturnRef;
+
+namespace {
+const std::string kVehicleData = "speed";
+}  // namespace
+
+class VehicleInfoPluginTest : public ::testing::Test {
+ public:
+  VehicleInfoPluginTest()
+      : vehicle_info_plugin_(nullptr)
+      , mock_message_helper_(
+            *application_manager::MockMessageHelper::message_helper_mock())
+      , applications_lock_(std::make_shared<sync_primitives::Lock>())
+      , applications_(application_set_, applications_lock_)
+      , mock_app_ptr_(nullptr) {
+    Mock::VerifyAndClearExpectations(&mock_message_helper_);
+  }
+
+  void SetUp() {
+    vehicle_info_plugin_ = std::make_shared<VehicleInfoPlugin>();
+    vehicle_info_plugin_->Init(app_manager_mock_,
+                               mock_rpc_service_,
+                               mock_hmi_capabilities_,
+                               mock_policy_handler_,
+                               nullptr);
+    mock_app_ptr_ =
+        std::make_shared<NiceMock<app_manager_test::MockApplication> >();
+  }
+
+  ~VehicleInfoPluginTest() {
```

Comment:
@ychernysheva how about to do the same stuff in tearDown?

Reply:
@mkorniichuk I've used destructor instead of tearDown(), because destructor is recommended to use in best practices, related to writing of unit tests, where MessageHelper or DataAccessor instances are used.